compat/maven-settings-builder/src/main/java/org/apache/maven/settings/building/DefaultSettingsBuilder.java

```diff
@@ -251,4 +255,29 @@ public Object execute(String expression, Object value) {
 
         return result;
     }
+
+    private List<Server> serversByIds(List<Server> servers) {
+
+        if (servers.stream().allMatch(server -> server.getIds().isEmpty())) {
+            return servers;
+        }
+
+        return servers.stream()
+                .flatMap(server -> {
+                    List<String> ids = server.getIds();
+                    if (ids.isEmpty()) {
+                        return Stream.of(server);
+                    } else {
+                        return Stream.concat(Stream.of(server), ids.stream().map(id -> newServer(server, id)));
+                    }
+                })
+                .collect(Collectors.toList());
+    }
+
+    private Server newServer(Server server, String id) {
+        return new Server(org.apache.maven.api.settings.Server.newBuilder(server.getDelegate(), false)
+                .id(id)
+                .ids(Collections.emptyList())
```

Comment:
```suggestion
                .ids(emptyList())
```
- https://docs.openrewrite.org/recipes/java/usestaticimport
- https://github.com/apache/maven/pull/2322